lib/bango/errors.py

```diff
@@ -53,3 +56,13 @@ class ProcessError(Exception):
 
     def __init__(self, response):
         self.response = response
+
+
+class StatusError(ParseError):
+
+    """
+    This is an error that occurs in the Stats checking code, which we do want
+    to commit so we've got a history of the transaction failure.
+    """
+
+    _do_not_rollback = True
```

Comment:
I don't think this is the right way to solve the problem.

Here's the problem:
* you want to raise an exception that indicates a failure
* everything written to the database is now questionable and should be rolled back because there was an exception (I know this doesn't apply to stats but I think in other cases it would)
* you want to commit some changes to the database *about the error*

I think this would be a better approach:

````
class StatusError:

    def report_error(self):
        self.obj.status = STATUS_BAD
        self.obj.errors = json.dumps({'form.errors': self.form_errors})
        self.obj.save()
````

The error raising code:
````
if not form.is_valid():
    raise StatusError(obj, form.errors)
````

The final error catching code:
````
def custom_exception_handler(exc):
    if hasattr(exc, 'report_errors'):
        transaction.rollback()
        exc.report_errors()
        set_rollback(False)
    else:
        set_rollback(True)  # let Django roll back
````
Without something like this, you'd never be able to roll back database changes **and** report errors.